apps/owmweather/lib.js

```diff
@@ -1,19 +1,20 @@
 function parseWeather(response) {
   let owmData = JSON.parse(response);
 
-  let isOwmData = owmData.coord && owmData.weather && owmData.main;
+  let isOwmData = (owmData.lat && owmData.lon) && owmData.current.weather && owmData.current;
 
   if (isOwmData) {
     let json = require("Storage").readJSON('weather.json') || {};
     let weather = {};
     weather.time = Date.now();
-    weather.hum = owmData.main.humidity;
-    weather.temp = owmData.main.temp;
-    weather.code = owmData.weather[0].id;
-    weather.wdir = owmData.wind.deg;
-    weather.wind = owmData.wind.speed;
-    weather.loc = owmData.name;
-    weather.txt = owmData.weather[0].main;
+    weather.hum = owmData.current.humidity;
+    weather.temp = owmData.current.temp;
+    weather.code = owmData.current.weather[0].id;
+    weather.wdir = owmData.current.wind_deg;
+    weather.wind = owmData.current.wind_speed;
+    weather.loc = owmData.name ? owmData.name : "";
```

Comment:
We can write this a little shorter if you prefer:
```suggestion
    weather.loc = owmData.name || "";
```

Similarly below with `.pressure`

Reply:
Implemented.